packages/docs/cookbook/migration-vuex.md

```diff
@@ -0,0 +1,271 @@
+    fullName: (state) => `${state.firstName} ${state.lastName}`,
+    loggedIn: (state) => !!state.userId,
+    // must define return type because of using `this`
+    fullUserDetails: (state): FullUserDetails => {
+      // import from other stores
+      const authPreferencesStore = useAuthPreferencesStore()
+      const authEmailStore = useAuthEmailStore()
+      return {
+        ...state,
+        // other getters now on `this`
+        fullName: this.fullName,
+        ...authPreferencesStore.$state,
+        ...authEmailStore.details
+      }
+
+      // alternative if other modules are still in Vuex
+      // return {
+      //   ...state,
+      //   fullName: this.fullName,
+      //   ...vuexStore.state.auth.preferences,
+      //   ...vuexStore.getters['auth/email'].details
+      // }
+    }
+  },
+  actions: {
+    // no context as first argument, use `this` instead
+    async loadUser (id: number) {
+      if (this.userId) throw new Error('Already logged in')
+      const res = await api.user.load(id)
+      this.updateUser(res)
+    },
+    // mutations can now become actions, instead of `state` as first argument use `this`
+    updateUser (payload) {
+      this.firstName = payload.firstName
+      this.lastName = payload.lastName
+      this.userId = payload.userId
+    },
+    // easily reset state using `$reset`
+    clearUser () {
+      this.$reset()
+    }
+  }
+})
+```
+
+Let's break the above down into steps:
+
+1. Add a required `id` for the store, you may wish to keep this the same as the namespace before
+2. Convert `state` to a function if it was not one already
+3. Convert `getters`
+    1. Remove any getters that return state under the same name (eg. `firstName: (state) => state.firstName`), these are not necessary as you can access any state directly from the store instance
+    2. If you need to access other getters, they are on `this` instead of using the second argument. Note you will need to specify a return type if you do this because of TS limitations, see [here](../core-concepts/getters.md#accessing-other-getters) for more details
+    3. If using `rootState` or `rootGetters` arguments, replace them by importing the other store directly, or if they still exist in Vuex then access them directly from Vuex
+4. Convert `actions`
+    1. Remove the first `context` argument from each action. Everything should be accessible from `this` instead
+    2. If using other stores either import them directly or access them on Vuex, the same as for getters
+5. Convert `mutations`
+    1. Mutations do not exist any more. These can be converted to `actions` instead, or you can just assign directly to the store within your components (eg. `userStore.firstName = 'First'`)
+    2. If converting to actions, remove the first `state` argument and replace any assignments with `this` instead
+    3. A common mutation is to reset the state back to its initial state. This is built in functionality with the store's `$reset` method
```

Comment:
I wonder if it should be pointed out somehow that the reset functionality only exists for option store (which isn't mentioned and I think that's normal)

Reply:
Added a note here and also at the start of this section stating that we're using option store as it's the most similar to Vuex.